prog/weaveproxy/main.go

```diff
@@ -15,6 +15,23 @@ var (
 	defaultListenAddrs = []string{"tcp://0.0.0.0:12375", "unix:///var/run/weave.sock"}
 )
 
+type listOpts struct {
+	values *[]string
```

Comment:
OK, you don't need the struct wrapper; just `type listOpts []string`.  That's what I did in https://github.com/weaveworks/weave/blob/1076-ignore-iface/prog/netcheck/netcheck.go